### PR TITLE
Fetch SearchHits scroll by scroll as needed than all at once in Elasticsearch

### PR DESCRIPTION
**Pull request by vkorukanti**
Currently we fetch all SearchHits in one call and keep them in memory. This causes
too much delay in getting first record and memory pressure. Write a custom iterator
that fetches the SearchHits as needed. One downside with this approach is the
scroll could get timedout, but there is a way to config custom scroll time in
plugin config.

### CODE REVIEW

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java [hunk 1]
```diff
@@ -334,4 +274,87 @@ public Object getValue()
...
+        }
+
+        @Override
+        protected SearchHit computeNext()
+        {
+            if (scrollId == null) {
+                // make the first request and get the scroll id
+                SearchResponse response = getSearchResponse(queryBuilder);
+
+                if (response.getHits().getTotalHits() > maxHits) {
```
Comment (martint) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java]: The limit made sense when we were buffering in memory, but do we need this now that we’re scrolling through results?

Comment (vkorukanti) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java]: Makes sense. Just pushed a commit that removes the check. Also marked the config as `Deprecated` (not sure if this is the correct way to handle config props that are no longer needed)

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchConnectorConfig.java [hunk 2]
```diff
@@ -105,13 +105,15 @@ public ElasticsearchConnectorConfig setScrollTimeout(Duration scrollTimeout)
         return this;
     }
 
+    @Deprecated
     @NotNull
     @Min(1)
     public int getMaxHits()
```
Comment (findepi) [inline presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchConnectorConfig.java]: `getMaxHits()` is now unused. I'd prefer to remove this config, adding this on top of the class:
```java
@DefunctConfig({
        "elasticsearch.max-hits",
})
```

If there is a strong reason not to remove this now, then let's have a TODO here and cleanup later (soonish).

File: presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java (hunk 1 above)
Comment (findepi) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java]: I'd prefer to remove this config, adding this on top of the class:
```java
@DefunctConfig({
        "elasticsearch.max-hits",
})
```

If there is a strong reason not to remove this now, then let's have a TODO here and cleanup later (soonish).

Comment (vkorukanti) [reply presto-elasticsearch/src/main/java/io/prestosql/elasticsearch/ElasticsearchRecordCursor.java]: Sure, added it to the `@DefunctConfig`.

**Comment by martint** [comment]
Merged, thanks!

**Pull request merged**